### DISCUSSION TITLE
What algorithms are used for hierarchical time series reconciliation  and CatBoostMultiSegmentModel

### DISCUSSION THREAD

**Discussion by Sandy4321** [etna-team/etna#476] [Q&A] [answered]
### 📖 Documentation improvement

pls how to find in documentation 
1
https://docs.etna.ai/1.15.1/reconciliation.html
what algorithms are used for hierarchical time series reconciliation
2

how CatBoostMultiSegmentModel is better than CatBoostPerSegmentModel 

and what algorithm used for CatBoostMultiSegmentModel 
or at least where in code CatBoostMultiSegmentModel  algorithm is coded

**Comment by d-a-bunin** [comment, marked as answer]
As for `CatBoostMultiSegmentModel` we already discussed it in a telegram chat. I'll repeat core messages for someone who could have the same question in the future.

The core ideas for understanding the `CatBoostPerSegmentModel` algorithm is contained [here](https://github.com/etna-team/etna/blob/0f98ec07865235683a1a7d537d4e3ceeff9fba70/etna/models/mixins.py#L533)
During fit
- For all the segments and all the timestamps generate a table there one line represent one timestamp for one segment with features calculated on segment history.
- Fit table with a given model

During forecast
- For all the segments and all the timestamps generate a table there one line represent one timestamp for one segment with features calculated on segment history.
- Predict table with a given model

For example, we have T timestamps in history, N segments and H horizon. The resulting table for learning catboost will have T * N rows (this is lower if we remove rows with missing values). The resulting table for predicting will have H * N rows.

We use a regular CatBoostRegressor, but for each point in the forecasting horizon there is a separate row in a table with its own features.
If you use Pipeline we call this a simplified direct forecasting strategy: https://docs.etna.ai/stable/tutorials/208-forecasting_strategies.html#3.1-Pipeline .

To determine which approach is better per-segment or multi-segment you should use backtest and look at the metrics. 
The apriori assumption is that for datasets with a lot of similar segments the multi-segment model is better
- It has more data to learn from
- A lot of segments share the same patterns, so the learned dependencies aren't very different from segment to segment
- It could still learn some difference between segments using some kind of segment encoding, e.g. with `SegmentEncoderTransform`.

**Comment by d-a-bunin** [comment]
As for hierarchical time series reconciliation, you are using a very old version of documentation. I advise to look at the documentation for the version you are currently using. Documentation for the latest available version is [here](https://docs.etna.ai/stable/index.html).

I think that good enough explanation about current implemented reconciliation methods is in [the tutorial about hierarchical time series](https://docs.etna.ai/stable/tutorials/303-hierarchical_pipeline.html).
